nifi-nar-bundles/nifi-extension-utils/nifi-record-utils/nifi-mock-record-utils/src/main/java/org/apache/nifi/serialization/record/MockRecordParser.java

```diff
@@ -58,6 +58,10 @@ public void addSchemaField(final String fieldName, final RecordFieldType type, b
         fields.add(new RecordField(fieldName, type.getDataType(), isNullable));
     }
 
+    public void addSchemaField(final RecordField recordField) {
```

Comment:
This preserves the full data type of the field. RecordFieldType.getDataType(), which is called from the other add() methods, returns the "base" type, such as "ARRAY" instead of "ARRAY[INT]", for equality purposes.